WooCommerce/Classes/ViewRelated/Themes/ThemesCarouselView.swift

```diff
@@ -0,0 +1,130 @@
+import SwiftUI
+import Kingfisher
+
+
+struct ThemesCarouselView: View {
+    /// Tuple array containing theme name and theme Url string pair.
+    let themesInfo: [(name: String, url: String)] = [
+        (name: "Google", url: "google.com"),
+        (name: "Facebook", url: "facebook.com"),
+        (name: "WordPress", url: "wordpress.com"),
+        (name: "Microsoft", url: "microsoft.com"),
+        (name: "Amazon", url: "amazon.com"),
+        (name: "Apple", url: "apple.com")
+    ]
+
+    /// The title of the message at the end of the carousel.
+    let lastMessageHeading: String
+
+    /// The content of the message at the end of the carousel.
+    let lastMessageContent: String
+
+
+    /// Scale of the view based on accessibility changes
+    @ScaledMetric private var scale: CGFloat = 1.0
+
+    var body: some View {
+        ScrollView(.horizontal, showsIndicators: false) {
+            HStack() {
+
+                // Demo for correct state
+                ForEach(themesInfo, id: \.name) { theme in
+                    if let themeUrl = getThemeUrl(themeUrl: theme.url) {
+                        themeImage(url: themeUrl)
+                    } else {
+                        themeNameCard(name: theme.name)
+                    }
+                }
+
+                // Demo for error state
+                themeNameCard(name: themesInfo[0].name)
+
+                // Message at the end of the carousel
+                lastMessageCard(heading: lastMessageHeading, message: lastMessageContent)
+            }
+        }
+    }
+}
+
+private extension ThemesCarouselView {
+    func themeImage(url: URL) -> some View {
+        KFImage(url)
+            .resizable()
+            .aspectRatio(contentMode: .fit)
+            .frame(width: Layout.imageWidth, height: Layout.imageHeight)
+            .cornerRadius(Layout.cornerRadius)
+            .shadow(radius: Layout.shadowRadius, x: 0, y: Layout.shadowYOffset)
+            .padding(Layout.imagePadding)
+    }
+
+    func themeNameCard(name: String) -> some View {
+        VStack {
+            Text(name)
+        }
+        .frame(width: Layout.imageWidth, height: Layout.imageHeight)
+        .background(Color.withColorStudio(
+            name: .wooCommercePurple,
+            shade: .shade0))
+        .cornerRadius(Layout.cornerRadius)
+        .shadow(radius: Layout.shadowRadius, x: 0, y: Layout.shadowYOffset)
+        .padding(Layout.imagePadding)
+    }
+
+    func lastMessageCard(heading: String, message: String) -> some View {
+        ZStack {
+            GeometryReader { geometry in
+                ScrollView(.vertical) {
+                    VStack {
+                        Spacer()
+                        Text(lastMessageHeading)
+                            .foregroundColor(Color(.secondaryLabel))
+                            .bold()
+                            .bodyStyle()
```

Comment:
Nit: this can be simplified with `secondaryBodyStyle`:

```suggestion
                            .bold()
                            . secondaryBodyStyle()
```

Reply:
Added in https://github.com/woocommerce/woocommerce-ios/pull/11336/commits/0e8687a65622534c196a1edca98825b822a59fb3